Jenks/add grpcrestapi reference to cometbft (#21)
babylonlabs-io/babylonlabs.github.io: docs/developer-guides/grpcrestapi.md

sidebar_label: gRPC Gateway (REST API)
hide_table_of_contents: true

---

# gRPC Gateway API 

> Babylond's implementation is based on CometBFT node rpcs. 
> In addition, some custom modules also allow Babylon specific methods. 

Reviewer: ```suggestion
The Babylon node implementation includes the built-in CometBFT node RPCs. 
In addition, Babylon implements custom modules that have dedicated RPC methods.
```

sidebar_label: gRPC Gateway (REST API)
hide_table_of_contents: true

---

# GRPC Gateway API 

> Babylond's implementation is based on CometBFT node rpcs. In addition, some custom modules also allow Babylon specific methods. 

Reviewer 2: Let's wrap lines to 80 characters on markdown files

> Please refer to both Babylon gRPC API and CometBFT API specifications for details. 

Reviewer 2: Could the below be a list?

> [Babylon Custom gRPC](https://app.swaggerhub.com/apis-docs/jvssptyltd/babylon_g_rpc_gateway_docs/1.0.0)

Reviewer 2: ```suggestion
[Babylon Application Layer gRPC](https://app.swaggerhub.com/apis-docs/jvssptyltd/babylon_g_rpc_gateway_docs/1.0.0)
```

> [Babylon Standard gRPC (CometBFT v0.83)](https://app.swaggerhub.com/apis-docs/jvssptyltd/comet-bft_rpc/v0.38.x)

Reviewer 2: ```suggestion
[Babylon Standard gRPC (CometBFT v0.38)](https://app.swaggerhub.com/apis-docs/jvssptyltd/comet-bft_rpc/v0.38.x)
```

Phase 3: Devnet Endpoints

| Provider      | RPC Endpoint                          | LCD Nodes Endpoint                     | gRPC Endpoint                          |
|---------------|---------------------------------------|---------------------------------------|----------------------------------------|

> | Babylon Labs  | https://rpc.devnet.babylonlabs.io    | https://lcd.devnet.babylonlabs.io    | https://grpc.devnet.babylonlabs.io    |

Reviewer: This is not a public devnet - suggest to either add Euphrates endpoints here or to include nothing at all.

Author: I will remove.

> Phase 2: Testnet Endpoints

Reviewer 2: Let's put testnet endpoints first?

> | Provider      | RPC Endpoint                          | LCD Nodes Endpoint                     | gRPC Endpoint                          |

Reviewer 2: How about we link to our [networks](https://github.com/babylonlabs-io/networks/tree/main/bbn-test-5) page to avoid having to update this if the URLs change?

Author: Okay, let me direct people there. I think it only makes sense to provide Phase 2 testnet endpoints for now as we don't expect a lot of activities on Phase 3 Devnet yet. 

Ideally, I would build a "network information" page to display all network parameters including RPC endpoints.
